tests/test_config.py

```diff
@@ -44,178 +68,100 @@ def test_config_adherence_for_code_inside_rules(self, cli_runner, tmp_path):
+            config.write_text(f"[tool.snakefmt]\nline_length = {line_length}\n")
+            params = ["--config", str(config), "-"]
 
-        expected_output = (
-            "rule a:\n"
-            f"{TAB*1}input:\n"
-            f"{TAB*2}list_of_lots_of_things=[\n"
-            f"{TAB*3}1,\n{TAB*3}2,\n{TAB*3}3,\n{TAB*3}4,\n{TAB*3}5,\n"
-            f"{TAB*3}6,\n{TAB*3}7,\n{TAB*3}8,\n{TAB*3}9,\n{TAB*3}10,\n"
-            f"{TAB*2}],\n"
-        )
+            actual = cli_runner.invoke(main, params, input=stdin)
 
-        assert actual.output == expected_output
+            assert actual.exit_code == 0
+            if expect_same:
+                assert actual.output == stdin
+            else:
+                assert actual.output != stdin
 
 
 class TestReadSnakefmtDefaultsFromPyprojectToml:
     def test_no_value_passed_and_no_pyproject_changes_nothing(self, testdir):
-        default_map = dict()
-        ctx = click.Context(click.Command("snakefmt"), default_map=default_map)
+        ctx = click.Context(click.Command("snakefmt"), default_map=dict())
         param = mock.MagicMock()
-        value = None
-
-        return_val = read_snakefmt_defaults_from_pyproject_toml(ctx, param, value)
 
+        return_val = inject_snakefmt_config(ctx, param, config_file=None)
         assert return_val is None
+        assert ctx.default_map == dict()
 
-        actual_default_map = ctx.default_map
-        expected_default_map = dict()
-
-        assert actual_default_map == expected_default_map
-
-    def test_pyproject_present_but_empty_changes_nothing_returns_pyproject_path(
-        self, testdir
-    ):
+    def test_empty_pyproject_is_detected_and_injects_nothing(self, cli_runner, testdir):
+        """ctx.params 'src' is set, simulating looking for pyproject.toml
+        in CLI passed formatted file's directory"""
         pyproject = Path("pyproject.toml")
         pyproject.touch()
-        default_map = dict()
-        ctx = click.Context(click.Command("snakefmt"), default_map=default_map)
+        ctx = click.Context(click.Command("snakefmt"))
+        ctx.params = dict(src=(str(Path().resolve()),))
         param = mock.MagicMock()
-        value = None
-
-        actual_config_path = read_snakefmt_defaults_from_pyproject_toml(
-            ctx, param, value
-        )
-        expected_config_path = str(pyproject)
-
-        assert actual_config_path == expected_config_path
+        actual_config_path = inject_snakefmt_config(ctx, param, None)
+        assert actual_config_path == str(pyproject.resolve())
         assert ctx.default_map == dict()
 
-    def test_no_value_passed_and_pyproject_present_changes_default_line_length(
-        self, testdir
-    ):
+    def test_nonempty_pyproject_is_detected_and_parsed(self, testdir):
         pyproject = Path("pyproject.toml")
-        pyproject.write_text("[tool.snakefmt]\nline_length = 4")
+        pyproject.write_text("[tool.snakefmt]\nline_length = 4\n" "foo = true")
```

Comment:
I feel uncomfortable with moving the testing of unknown params into another test. This test is now testing two different behaviours but only states that it is testing one thing. Silent behaviour is not great for testing. Makes it harder to tweak things in the future.

Reply:
What is special about `foo = true` as compared to `line_length` which we actually use? I moved it here because unknown params under [tool.snakefmt] do not get any special treatment in our  code currently. If we for eg raised an error because `foo` is not a CLI param, i would agree with having it being its own test.

Note this is in contrast with `test_unrecognised_black_options_in_config_ignored_and_default_line_length_used` where unrecognised params are treated specially (ignored), and is its own test.

Reply:
Again, it is more a future-proofing mechanism. Whilst we don't explicitly treat unknowns specially now, we may change some behaviour in the future that unexpectedly also changes how unknowns are handled. If we have a test that is keeping an eye on how unknowns are handled, we don't have to worry about breaking changes, as we will have tests that alert us to this first.

Reply:
Feel free to propose a change of this, I think that any feature should also have a test with it, to me it's a bloat to make tests 'in case we make a future feature' especially given the current way it works seems good.

Reply:
> Feel free to propose a change of this

But you're the one proposing a change to this (in this PR) and I am saying I do'nt agree with the change. i.e. I am proposing keeping the current state.

Reply:
Got it. i'm re-adding a test for parsing unknown params.